### PR TITLE
javadoc: Allow `javadoc` targets to reference other `javadoc` targets

### PR DESCRIPTION
**Pull request by fmeum**
By setting the `doc_url` attribute on a `javadoc` target, other targets can depend on it via `doc_deps` and have references resolved automatically via appropriate `-linkoffline` arguments passed to javadoc.

### CODE REVIEW

**Comment by fmeum** [comment]
@shs96c Happy to bikeshed any of the new API bits. I briefly considered merging `doc_deps` into `deps`, but that felt too magical.

**Comment by shs96c** [comment]
I'm about to head to TPAC soon, so it's going to take a few days to get to do a proper review of this, but it seems fine from a quick scan. I do appreciate minimal magic in rulesets, so not merging `doc_deps` and `deps` is a good choice to have made.

**Review by shs96c — APPROVED** [review]
Seems reasonable to me. Would have been nice to have a test, but we can add that in a later PR.

File: private/rules/java_export.bzl [hunk 1]
```diff
@@ -113,7 +121,10 @@ def maven_export(
         tags = [],
         testonly = False,
         javadocopts = [],
-        classifier_artifacts = {}):
+        classifier_artifacts = {},
+        *,
```
[the remark this answers is not in the document]
Comment (fmeum) [reply private/rules/java_export.bzl:125]: This syntax enforces that all arguments that come after it are only passed as keyword arguments, not positional arguments. I usually use it for functions that have more than say two keyword parameters, just to guard against accidentally passing in unexpected arguments positionally.

**Pull request merged**